### ISSUE TITLE
not found should return 404

### ISSUE THREAD

**Issue by Nemo64** [nekudo/shiny_geoip#20]
When I request http://geoip.nekudo.com/api/127.0.0.1 then I get an error which is correct, but the http status is 200. The http status should be 404.

**Comment by nekudo** [comment]
Hi.

This one is a little tricky. While I totally get your point I decided not to implement this. The reason:
The IP address is an optional parameter. This means the URL http://geoip.nekudo.com/api could respond with either 200 or 404 depending on the clients IP. This however would not be correct in the sense of the 404 status as the resource /api is available.

Hopefully this is understandable.

Regrards
Simon

**Comment by Nemo64** [comment]
Well, I kind of get your point but 404 doesn't say that the url is invalid, it says that the resource is unavailable, which it isn't for that person.

But even if you dislike 404, i think it is closer to correct than 200 Ok.

It's also more annoying while implementing it since you'll need to specially check if there is an error even when the response was a success. So this simple js implementation (which did work with freegeoip.net) is now flawed (pseudo implementation)
```JavaScript
fetch('http://geoip.nekudo.com/api')
    .then(response => console.log('success', response))
    .catch(error => console.error('error', error))
```
I now need something likes this
```JavaScript
fetch('http://geoip.nekudo.com/api')
    .then(response => response.type === 'error' ? Promise.reject(response) : response)
    .then(response => console.log('success', response))
    .catch(error => console.error('error', error))
```
That is neither intuitive nor documented on your start page https://geoip.nekudo.com

There are also some alternatives if you dislike 404, like 400 Bad Request or even 500 would be better imo. I'd even go as far and say 418 is better than 200 because at least it's an error code.

**Comment by nekudo** [comment]
Fair point. I'll give this another thought. (As soon as I can spare a few hours)
